test/test_mypy.py

```diff
@@ -304,6 +324,62 @@ def test_typeddict_document_type(self) -> None:
         assert retreived["year"] == 1
         assert retreived["name"] == "a"
 
+    @only_type_check
+    def test_typeddict_document_type_insertion(self) -> None:
+        client: MongoClient[Movie] = MongoClient()
+        coll: Collection[Movie] = client.test.test
+        mov = Movie(_id=ObjectId(), name="THX-1138", year=1971)
+        for meth, arg in [
+            (coll.insert_many, [mov]),
+            (coll.insert_one, mov),
+            (coll.bulk_write, [InsertOne(mov)]),
+        ]:
+            meth(arg)  # type:ignore[operator]
```

Comment:
How can we put a type ignore here? insert_many/insert_one/bulk_write working without raising a mypy error is the main purpose of this bug fix. I think we're loosing sight of what's important here because the PR is muddled by tests for find_one/_id/NotRequired. Can you open a new ticket for the NotRequired stuff? To fix the bug in PYTHON-3454 I believe we only need to mention schema validation should be configured on the server and add a test like this to confirm the bug is fixed:
```python
    def test_typeddict_document_type_insertion(self) -> None:
        coll: Collection[Movie] = self.db.test
        mov = Movie(name="THX-1138", year=1971)
        coll.insert_one(mov)
        coll.insert_many([mov])
        coll.bulk_write([InsertOne(mov)])
```

Reply:
Let's remove everything about _id, NotRequired, typing_extensions, and all the other tests besides the one above. All that is extra stuff that's only tangentially related to the issue. We already have tests that find_one correctly yields the collection generic type: https://github.com/mongodb/mongo-python-driver/blob/4.3.2/test/test_mypy.py#L299-L305

Reply:
There are two complaints in [PYTHON-3454](https://jira.mongodb.org/browse/PYTHON-3454):
1) `collection.insert_one(Person({"name": "Harold", "age": 15}))` incorrectly raises a mypy error. This is fixed by using _DocumentType and tested by `test_typeddict_document_type_insertion` above.
2) `collection.insert_one({"name": 17, "age": "this isn't a number", "gender": 6})` does not raise a mypy error when it should. This is also fixed by using  _DocumentType and should also be tested (if we aren't doing it already).

This is why I see the _id, NotRequired, typing_extensions, and find test as all unrelated. They are independent from the 2 issues reported by the user.

Reply:
> Perhaps I am misunderstanding, but I don't think that the user wanted to be able to insert a dict and have that be type-checked.

They do want this. In fact that's all a TypeDict is, it's a regular dict that's been type checked. So both of these inserts are completely identical and should have the same behavior:
```python
collection: Collection[Person] = client.t.t
collection.insert_one(Person({"name": "Harold", "age": 15})) 
collection.insert_one({"name": "Harold", "age": 15}) 
```
And both of these line should also cause the same mypy error:
```python
collection: Collection[Person] = client.t.t
collection.insert_one(Person({"name": "Harold", "age": "WRONG TYPE"})) 
collection.insert_one({"name": "Harold", "age": "WRONG TYPE"}) 
```
Both of these issues will be fixed by this PR. 

I agree that the subtleties of _id + TypedDict are worth documenting but I would like you to open a new jira ticket for it that we can prioritize separately. It's a different issue from PYTHON-3454.

Reply:
In the original example the `TypeDict` did not have an explicit `_id` field.

Reply:
Not including the `_id` is still a valid choice, if it is not relevant to your application.